specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx

```diff
@@ -211,67 +247,103 @@ export function IntegratedRecordSelector({
         isLoading,
       }): JSX.Element => (
         <>
-          <ResourceView
-            dialog={dialog}
-            headerButtons={(specifyNetworkBadge): JSX.Element => (
-              <>
-                <DataEntry.Visit
-                  /*
-                   * If dialog is not false, the visit button would be added
-                   * by ResourceView
-                   */
-                  resource={
-                    !isDependent && dialog === false ? resource : undefined
-                  }
-                />
-                {hasTablePermission(
-                  relationship.relatedModel.name,
-                  isDependent ? 'create' : 'read'
-                ) && typeof handleAdd === 'function' ? (
-                  <DataEntry.Add
-                    disabled={
-                      mode === 'view' ||
-                      (isToOne && collection.models.length > 0)
+          {typeof recordSetsPromise === 'object' && isInteraction ? (
+            <InteractionDialog
+              action={{
+                model: collection.related?.specifyModel as SpecifyModel<
+                  Disposal | Gift | Loan
```

Comment:
Moving the definition of `isInteraction` to this component might allow to simplify this typing (and simplify the definition of isInteraction).

<details><summary>Solution</summary>
<p>



```tsx
const interaction = typeof collection.related === 'object' ? (
  toTable(collection.related,'LoanPreparation') ??
  toTable(collection.related,'GiftPreparation') ??
  toTable(collection.related,'DisposalPreparation')
) : undefined;
```
</p>
</details>